web/packages/teleport/src/AuthConnectors/AuthConnectorEditor/AuthConnectorEditorContent.tsx

```diff
@@ -0,0 +1,141 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Link as RouterLink } from 'react-router-dom';
+
+import { Link } from 'design';
+import { Alert } from 'design/Alert';
+import Box from 'design/Box';
+import { ButtonPrimary, ButtonSecondary } from 'design/Button';
+import Flex from 'design/Flex';
+import { ArrowBack } from 'design/Icon';
+import { Indicator } from 'design/Indicator';
+import { H1, H3 } from 'design/Text';
+import { P } from 'design/Text/Text';
+import TextEditor from 'shared/components/TextEditor';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+
+import { DesktopDescription } from 'teleport/AuthConnectors/styles/AuthConnectors.styles';
+import { FeatureBox, FeatureHeaderTitle } from 'teleport/components/Layout';
+
+import { description } from '../AuthConnectors';
+
+/**
+ * AuthConnectorEditorContent is a the content of an Auth Connector editor page.
+ */
+export function AuthConnectorEditorContent({
+  title,
+  content,
+  backButtonRoute,
+  isSaveDisabled,
+  saveAttempt,
+  fetchAttempt,
+  onSave,
+  onCancel,
+  setContent,
+}: Props) {
+  return (
+    <FeatureBox>
+      <FeatureHeaderTitle py={3} mb={2}>
+        <Flex alignItems="center">
+          <ArrowBack
+            as={RouterLink}
+            mr={2}
+            size="large"
+            color="text.main"
+            to={backButtonRoute}
+          />
+          <Flex mr={4} alignItems="baseline">
+            <H1>{title}</H1>
+          </Flex>
+        </Flex>
+      </FeatureHeaderTitle>
+      {fetchAttempt.status === 'failed' && (
+        <Alert>{fetchAttempt.statusText}</Alert>
+      )}
+      {fetchAttempt.status === 'processing' && (
+        <Box textAlign="center" m={10}>
+          <Indicator />
+        </Box>
```

Comment:
if this margin is used to center, you could maybe use Flex with some justify and align instead of adding a bunch of margin. doesnt really matter tho